src/bootstrap/compile.rs

```diff
@@ -1108,6 +1108,11 @@ impl Step for Assemble {
             let src_exe = exe("lld", target_compiler.host);
             let dst_exe = exe("rust-lld", target_compiler.host);
             builder.copy(&lld_install.join("bin").join(&src_exe), &libdir_bin.join(&dst_exe));
+            // for gcc-lld linker flavor
+            builder.copy(
+                &lld_install.join("bin").join(&src_exe),
+                &libdir_bin.join(exe("ld", target_compiler.host)),
+            );
```

Comment:
`src\bootstrap\dist.rs` has analogous logic of copying `rust-lld` to `lib/rustlib/<host>/bin`, this bit should probably added there as well.